stingray/utils.py

```diff
@@ -189,11 +189,14 @@ def rebin_data(x, y, dx_new, yerr=None, method='sum', dx=None):
     """
 
     y = np.asarray(y)
-    yerr = np.asarray(apply_function_if_none(yerr, y, np.zeros_like))
+    if yerr is None:
+        yerr = np.zeros_like(y)
+    else:
+        yerr = np.asarray(yerr)
 
-    if not dx:
+    if dx is None or not dx:
```

Comment:
In theory yes, but I had a problem during tests with Numba that seemed to indicate otherwise. Anyways, I'm not using Numba in the end so I'll remove it